sdk/core/azure-core/src/main/java/com/azure/core/http/rest/SwaggerMethodParser.java

```diff
@@ -273,21 +274,33 @@ public String setPath(Object[] methodArguments) {
      * @param swaggerMethodArguments the arguments that will be used to create the query parameters' values
      * @param urlBuilder The {@link UrlBuilder} where the encoded query parameters will be set.
      */
+    @SuppressWarnings("unchecked")
     public void setEncodedQueryParameters(Object[] swaggerMethodArguments, UrlBuilder urlBuilder) {
         if (swaggerMethodArguments == null) {
             return;
         }
 
-        for (Substitution substitution : querySubstitutions) {
+        for (QuerySubstitution substitution : querySubstitutions) {
             final int parameterIndex = substitution.getMethodParameterIndex();
             if (0 <= parameterIndex && parameterIndex < swaggerMethodArguments.length) {
                 final Object methodArgument = swaggerMethodArguments[substitution.getMethodParameterIndex()];
-                String parameterValue = serialize(serializer, methodArgument);
-                if (parameterValue != null) {
-                    if (substitution.shouldEncode()) {
-                        parameterValue = UrlEscapers.QUERY_ESCAPER.escape(parameterValue);
+                List<Object> methodArguments = new ArrayList<Object>();
+
+                if (methodArgument instanceof List && substitution.shouldMergeQueryParams()) {
```

Comment:
From a perf standpoint I would switch these two conditions of the `if` statement to run the cheap boolean check first and only if it is true run the `instanceof` check. I would also suggest making this code more performant by removing the intermediate `methodArguments` list and just having two code paths in these if / else code block.